packages/back-end/src/scim/users/listUsers.ts

```diff
@@ -35,94 +37,27 @@ import { ScimListRequest } from "../../../types/scim";
 // };
 
-function isEmailAddress(input: string): boolean {
-  const emailRegex = /^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}$/;
-  return emailRegex.test(input);
-}
-
-function parseQueryFilter(queryFilter?: string) {
-  const regexPattern = /(\w+) (\w+) "([^"]+)"/;
-
-  let match;
-
-  if (queryFilter) {
-    match = regexPattern.exec(queryFilter);
-  }
-
-  // Check if there is a match
-  if (match) {
-    const [, filterBy, operator, value] = match; // Destructuring the matched values
-    const results = {
-      filterBy,
-      operator,
-      value,
-    };
-    return results;
-  }
-}
 //TODO: There is something wrong with the listUsersValidator - it returns an error "message": "Unexpected token o in JSON at position 1"
-export async function listUsers(req: ScimListRequest, res: Response) {
+export async function listUsers(
+  req: Request & ApiRequestLocals,
+  res: Response
+) {
   console.log("listUsers endpoint hit");
-  //TODO: Without the validator, the line below isn't happy because filter is possibly undefined
-  const queryInfo = parseQueryFilter(req.query.filter);
-
-  const isFilterValueEmail = isEmailAddress(queryInfo?.value);
-
-  console.log("isFilterValueEmail:", isFilterValueEmail);
-
-  //TODO: Update this to be more dynamic
-  const filterValue = queryInfo?.value;
-
-  console.log("filterValue", filterValue);
-
-  if (!filterValue) {
-    return {
-      schemas: ["urn:ietf:params:scim:api:messages:2.0:ListResponse"],
-      totalResults: 0,
-      Resources: [],
-      startIndex: 1,
-      itemsPerPage: 20,
-    };
-  }
+  const { startIndex, count, filter: filterQuery } = req.query;
 
   const org = req.organization;
 
-  console.log("org", org.id);
-
-  // const user = await getUserByEmail(userEmail);
-  let user: UserInterface | null;
-
-  //TODO: This should actually return a list of users, not just one
-  if (isFilterValueEmail) {
-    user = await getUserByEmail(filterValue);
-  } else {
-    user = await getUserByExternalId(filterValue);
-  }
-
-  console.log("user", user);
-
-  //TODO: We need to loop through all users and only return those users who are a member of this org
-  const orgUser = org.members.find((member) => member.id === user?.id);
-
-  console.log("orgUser", orgUser);
+  const expandedMembers = await expandOrgMembers(org.members);
 
-  if (!user || !orgUser) {
-    return res.status(200).json({
-      schemas: ["urn:ietf:params:scim:api:messages:2.0:ListResponse"],
-      totalResults: 0,
-      Resources: [],
-      startIndex: 1,
-      itemsPerPage: 20,
-    });
-  }
-
-  const resourcesToReturn = [
-    {
+  const scimUsers = expandedMembers.map((user) => {
+    return {
       schemas: ["urn:ietf:params:scim:schemas:core:2.0:User"],
-      id: user.externalId,
-      userName: user.externalId,
+      id: user.id,
+      userName: user.email,
       name: {
         displayName: user.name,
+        givenName: user.name.split(" ")[0],
+        familyName: user.name.split(" ")[1],
```

Comment:
This might give an index out of bounds error if `user.name` has no spaces for some reason. open to suggestions